### PR TITLE
[SofaBaseTopology] Some cleaning in BaseTopologyData and TopologyData

### PR DESCRIPTION
**Pull request by epernod**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/TriangleLocalMinDistanceFilter.cpp [hunk 1]
```diff
@@ -311,7 +312,8 @@ bool TriangleLocalMinDistanceFilter::validPoint(const Index pointIndex, const de
 
 bool TriangleLocalMinDistanceFilter::validLine(const Index lineIndex, const defaulttype::Vector3 &PQ)
 {
-    LineInfo &Li = m_lineInfo[lineIndex];  // filter is precomputed
+    helper::WriteAccessor< Data<sofa::helper::vector<LineInfo> > > lInfo(m_lineInfo);
```
Comment (fredroy) [inline SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/TriangleLocalMinDistanceFilter.cpp:312]: `auto lInfo = sofa::helper::getWriteAccessor(m_lineInfo)` is clearer

**Comment by guparan** [comment]
```text
/builds/workspace/sofa-framework/PR-1980/ubuntu_gcc-7_options_release/src/SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyData.inl:91:57: error: 'class sofa::component::topology::TopologyData<unsigned int, sofa::type::vector<double, sofa::type::CPUMemoryManager<double> > >' has no member named 'm_topology'
         this->m_topologyHandler->registerTopology(this->m_topology);
                                                   ~~~~~~^~~~~~~~~~
/builds/workspace/sofa-framework/PR-1980/ubuntu_gcc-7_options_release/src/SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyData.inl:92:21: error: 'class sofa::component::topology::TopologyData<unsigned int, sofa::type::vector<double, sofa::type::CPUMemoryManager<double> > >' has no member named 'm_topology'
     else if (!this->m_topology)
               ~~~~~~^~~~~~~~~~              
```

File: SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/TriangleLocalMinDistanceFilter.cpp (hunk 1 above)
Comment (epernod) [reply SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/TriangleLocalMinDistanceFilter.cpp:312]: If I apply this change I get this error:
```
TriangleLocalMinDistanceFilter.cpp(317,35): error C2676: binary '[': 'sofa::helper::WriteAccessor<sofa::component::topology::TopologyData<sofa::topology::Edge,sofa::type::vector<sofa::component::collision::LineInfo,sofa::type::CPUMemoryManager<T>>>,void>' does not define this operator or a conversion to a type acceptable to the predefined operator
12>        with
12>        [
12>            T=sofa::component::collision::LineInfo
12>        ]
12>C:\projects\sofa-src\SofaKernel\modules\SofaMeshCollision\src\SofaMeshCollision\TriangleLocalMinDistanceFilter.cpp(317,18): error C2530: 'Li': references must be initialized
```

Comment (fredroy) [reply SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/TriangleLocalMinDistanceFilter.cpp:312]: Ah, too bad; no choice to keep as it is then.

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Comment by epernod** [comment]
Changes from this PR will be included in other PR starting from #2082

**Pull request closed without merging**